### DISCUSSION TITLE
Access default actionbar back button Android

### DISCUSSION THREAD

**Discussion by Andarius** [wix/Detox#4022] [Q&A]
Hi! 
I'm trying to access the back button of `react-native-screens` which is set by calling `setDisplayHomeAsUpEnabled` (https://github.com/software-mansion/react-native-screens/blob/46794e840b9ee06008732c36b3bda61c18fe93c4/android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.kt#L183).
My understanding is that to access the back button I should try to access `android.R.id.home`.
Is it something that is exposed by Detox? Or is there another way?

**Comment by portyu9** [comment]
Hi @Andarius 

`by.id()` in Detox is not a matcher for arbitrary Android resource IDs such as:

```java
android.R.id.home
```

It normally matches the application's Detox/RN `testID` representation.

If what you want to test is the **navigation behavior**, the most robust Android-specific option is:

```js
await device.pressBack()
```

Detox still exposes `device.pressBack()` specifically for Android.

If you specifically need to tap the visible ActionBar "Up" button instead of performing the system back action, try matching it through its accessibility/content description with `by.label(...)` if that control is exposed.

Be careful with hard-coded labels such as `"Navigate up"` because they can vary with localization/platform behavior.

If the header is under your application's control, adding your own explicit accessibility identifier or `testID` is preferable.

So Detox does not provide a normal `by.androidResourceId(android.R.id.home)` equivalent; `device.pressBack()` is generally the better choice when you are testing back-navigation semantics.

Please mark this answer as accepted if it helped, thank you.
